### PR TITLE
Wconstab/shape2

### PR DESCRIPTION
**Pull request by wconstab**

### CODE REVIEW

File: torch/csrc/lazy/ts_backend/ts_node.cpp [hunk 1]
```diff
@@ -18,17 +18,11 @@ namespace torch {
 namespace lazy {
 
 const Shape& GetShapeFromTsOutput(const Output& output) {
-  if (auto* tsnode = dynamic_cast<const TsNode*>(output.node)) {
-    return tsnode->shape(output.index);
-  }
-  throw std::runtime_error("Expected TsNode but could not dynamic cast");
+  return output.node->shape(output.index);
```
Comment (antoniojkim) [inline torch/csrc/lazy/ts_backend/ts_node.cpp:21]: Can't this just be
```suggestion
  return output.shape();
```
also, is this function even used anymore? Can it be removed?

File: torch/csrc/lazy/ts_backend/ts_node.cpp [hunk 2]
```diff
@@ -18,17 +18,11 @@ namespace torch {
...
 }
 
 const Shape& GetShapeFromTsValue(const Value& value) {
-  if (auto* tsnode = dynamic_cast<const TsNode*>(value.node.get())) {
-    return tsnode->shape(value.index);
-  }
-  throw std::runtime_error("Expected TsNode but could not dynamic cast");
+  return value.node->shape(value.index);
```
Comment (antoniojkim) [inline torch/csrc/lazy/ts_backend/ts_node.cpp:25]: same here

File: torch/csrc/lazy/ts_backend/ts_node.cpp (hunk 1 above)
Comment (antoniojkim) [reply torch/csrc/lazy/ts_backend/ts_node.cpp:21]: In fact, it looks like you removed it from `ts_node.h` altogether. So, I think this definition can be removed

**Review by antoniojkim — APPROVED** [review]
Looks good to me!

**Comment by antoniojkim** [comment]
@wconstab what's the status on this PR? any ideas when it might be able to be merged in?

**Comment by wconstab** [comment]
oh, sorry- I merged it already -  but I forgot I had created a second PR so I merged the first PR with the same contents as shape2

**Pull request closed without merging**